pkg/services/ngalert/api/authorization.go

```diff
@@ -125,20 +125,34 @@ func (api *API) authorize(method, path string) web.Handler {
 	// Silences. External AM.
 	case http.MethodDelete + "/api/alertmanager/{DatasourceID}/api/v2/silence/{SilenceId}":
 		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceID")))
+	case http.MethodDelete + "/api/alertmanager/uid/{DatasourceUID}/api/v2/silence/{SilenceId}":
+		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceUID")))
 	case http.MethodPost + "/api/alertmanager/{DatasourceID}/api/v2/silences":
 		eval = ac.EvalPermission(ac.ActionAlertingInstancesExternalWrite, datasources.ScopeProvider.GetResourceScope(ac.Parameter(":DatasourceID")))
+	case http.MethodPost + "/api/alertmanager/uid/{DatasourceUID}/api/v2/silences":
```

Comment:
If you're up for it, it would be great to make a path function that at least provides the prefix:
```go
AMV2Path(pathSuffix) string {
  return path.Join("/api/alertmanager/datasource/{DatasourceUID}/api/v2", pathSuffix)
}

...

AMV2Path("silence/{SIlenceID")
```

that makes it easier to scan which API is being evaluated.

If not, I'll create an issue and I can tackle it later.